client/components/SearchOperationList.js

```diff
@@ -24,8 +24,13 @@ export default class SearchComponent extends React.Component {
         }
     }
 
-    clearSearch() {
-        this.setState(this.initialState(), this.filter);
+    clearSearch(close) {
+        let initialState = this.initialState();
+
+        if (!close)
+            delete initialState.showDetails;
```

Comment:
As a JavaScript VM developer, I have to warn you: never, in any case, **NEVER** `delete` a property on an object instance, unless you're willing to kill performance (if you want to learn why, search "hidden classes javascript virtual machine" on any search engine).

How about, instead, you `bind` the second argument of `clearSearch` to `true` for the "clear & close" button, and here just set `initialState.showDetails = !close;` ?